src/views/components/UserProfile/index.tsx

```diff
@@ -0,0 +1,221 @@
+import React from "react";
+import {
+  Button,
+  Container,
+  Form,
+  Input,
+  Message,
+  Select
+} from "semantic-ui-react";
+import options from "./helper";
+
+export interface Opt {
+  key: string;
+  text: string;
+  value: string;
+}
+
+export interface LoggedIn {
+  userName: string;
+  role: string;
+}
+
+export interface Profile {
+  userName: string;
+  password: string;
+  email: string;
+  role: string;
+}
+
+export interface Props {
+  roleOptions: Opt[];
+  editUser: Profile;
+  loggedInUser: LoggedIn;
+  autoFocus: boolean;
+  onSubmit: (data: any) => Promise<void>;
+  search: boolean;
+  selection: boolean;
+  options: Opt[];
+  serviceOptions: Opt[];
+  branches: Opt[];
+  onChange: (e) => void;
+  deleteUserHandler: (e) => void;
+}
+
+export interface State {
+  editUser: Profile;
+  emailError?: string;
+}
+
+class UserProfile extends React.Component<Props, State> {
+  public state: State;
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      editUser: this.props.editUser,
+      emailError: null
+    };
+  }
+
+  public handleInput = event => {
+    const name = event.target.name;
+    const value = event.target.value;
+
+    const editUser = { ...this.state.editUser, [name]: value };
+    this.setState({ editUser });
+  };
+  // TODO: Method to be used in dropdown though do not how to get the event.target.value
+  // public handleDropdownInput = (event, { value }) => {
+  //   const name = event.target.name;
+
+  //   const editUser = { ...this.state.editUser, [name]: value}
+  //   this.setState({ editUser });
+  // }
+
+  public validate = (): boolean => {
+    if (!this.state.editUser.email.toLowerCase().endsWith("@nssfug.org")) {
+      const emailError = "Please provide an nssfug.org email";
+      this.setState({ emailError });
+      return false;
+    }
+    this.setState({ emailError: null });
+    return true;
+  };
+
+  public handleSubmit = async () => {
+    if (this.validate()) {
+      await this.props.onSubmit(this.state.editUser);
+    }
+  };
+
+  public deleteUser = (role: string) => {
+    return role === "Admin" ? (
+      <Button
+        type="submit"
+        style={{ marginTop: "10px" }}
+        onClick={this.props.deleteUserHandler}
+      >
+        Delete Agent
+      </Button>
+    ) : (
+      <span />
+    );
+  };
+
+  public adminField = (role: string) => {
+    return role === "Admin" ? (
+      <span>
+        <Form.Field>
+          <label>Roles</label>
+          <Select
+            placeholder="role"
+            name="role"
+            options={options(
+              this.props.loggedInUser.role,
+              this.props.roleOptions
+            )}
+            // value={this.state.editUser.role}
```

Comment:
Select tags in Semantic UI are dropDown elements. You set the default value with `defaultValue` prop.
```suggestion
            defaultValue={this.state.editUser.role}
```